### PR TITLE
Add builtin typedefs

### PR DESCRIPTION
**Pull request by mfedderly**

### CODE REVIEW

**Comment by rowanwins** [comment]
Thanks @mfedderly ! Published as 1.4.0

File: types.d.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { FeatureCollection, Feature, GeometryObject } from "geojson";
+
+export type Intersection = [number, number];
+
+export default function sweeplineIntersections(
+  geojson: FeatureCollection<GeometryObject> | Feature<GeometryObject>,
```
Comment (twelch) [inline types.d.ts:6]: @mfedderly perhaps you had reasons, but should should geojson param also accept GeometryObject, not just Feature and FeatureCollection.  Also it's not all GeometryObject types according to the docs?, just `LineString | MultiLineString | Polygon | MultiPolygon`.  So the signature would be something like:

```
geojson: FeatureCollection<LineString | MultiLineString | Polygon | MultiPolygon> | Feature<LineString | MultiLineString | Polygon | MultiPolygon> | LineString | MultiLineString | Polygon | MultiPolygon
```

See docs:
"Valid inputs: Geojson Feature or Geometry including Polygon, LineString, MultiPolygon, MultiLineString, as well as FeatureCollection."

Comment (twelch) [reply types.d.ts:6]: The second argument `ignoreSelfIntersection` also should be optional according to the docs

Comment (rowanwins) [reply types.d.ts:6]: You are right @twelch 😂  - my review clearly stank!

Comment (mfedderly) [reply types.d.ts:6]: good catch @twelch !

**Pull request merged**